core/control-plane/control-plane-contract/src/main/java/org/eclipse/edc/connector/controlplane/contract/policy/ContractExpiryFunction.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright (c) 2024 Cofinity-X
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Cofinity-X - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.controlplane.contract.policy;
+
+import org.eclipse.edc.connector.controlplane.contract.spi.policy.TransferProcessPolicyContext;
+import org.eclipse.edc.connector.controlplane.policy.contract.ContractExpiryCheck;
+import org.eclipse.edc.policy.engine.spi.AtomicConstraintRuleFunction;
+import org.eclipse.edc.policy.model.Operator;
+import org.eclipse.edc.policy.model.Permission;
+
+/**
+ * {@link ContractExpiryCheck} wrapper to be used with {@link TransferProcessPolicyContext}.
+ */
+public class ContractExpiryFunction implements AtomicConstraintRuleFunction<Permission, TransferProcessPolicyContext> {
```

Comment:
nit: I find this name slightly misleading, because it does not really state the purpose clearly. `ContractExpiryFunction` could also imply a "function to expire a contract". How about `ContractExpiryCheckFunction`? this would also align it with `ContractExpiryCheck`